app/src/org/commcare/android/database/user/UserDatabaseUpgrader.java

```diff
@@ -242,4 +311,45 @@ private void markSenseIncompleteUnsent(final SQLiteDatabase db) {
             storage.write(t);
         }
     }
+
+    private static boolean multipleInstalledAppRecords() {
+        int globalDbVersionNumber = CommCareApplication._().getGlobalDbVersion();
+        if (globalDbVersionNumber >= 3) {
+            // If the 1st multiple apps migration has already occurred, then app records will be
+            // stored in the db as an ApplicationRecord
+            SqlStorage<ApplicationRecord> storage =
+                    CommCareApplication._().getGlobalStorage(ApplicationRecord.class);
+            int count = 0;
+            for (ApplicationRecord r : storage) {
+                if (r.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
+                    count++;
+                }
+            }
+            return (count > 1);
+        } else {
+            // If the 1st multiple apps migration has NOT already occurred, then app records will
+            // be stored in the db as an ApplicationRecordV1
+            SqlStorage<ApplicationRecordV1> storage =
+                    CommCareApplication._().getGlobalStorage(ApplicationRecordV1.class);
+            int count = 0;
+            for (ApplicationRecordV1 r : storage) {
+                if (r.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
+                    count++;
+                }
+            }
+            return (count > 1);
+        }
+    }
+
+    public static ApplicationRecord getInstalledAppRecord() {
+        SqlStorage<ApplicationRecord> storage =
```

Comment:
Yeah, migrations occur when the DB is opened for the first time, and we need to open the Global and App DB's to get the UserDB handles, so you should feel comfortable assuming that any Global DB migrations have already commited.